Burkhard_Keller_Tree/burkhard_keller_tree.cpp

```diff
@@ -0,0 +1,162 @@
+/* 
+
+Burkhard Keller Tree is a popular String Matching Algorithm which is used to
+perform spell checks based on the concept of Levenshtein distance.
+
+*/
+#include <iostream>
+#include <vector>
+#include <map>
+typedef std::vector< std::string > strlist;
+// Defining the Node 
+class Node {
+    public:
+        std::map< int, Node* > TreeEdge;
+        std::string Name;
+};
+// Defining the BK Tree
+class BKTree {
+    public:
+        strlist search(std::string Name, int target_dist);
+        void insert_into(strlist& list);
+
+        BKTree(std::string& Root_STR);
+        ~BKTree();
+
+    private:
+        Node* root;
+        void insert(std::string& Name, Node*& node);
+        void search(Node* node, std::string& Name, int target_dist, strlist& suggestion_points);
+        int min(int& a, int& b, int& c);
+        int d(std::string& test, std::string& comp);
+        void delete_node(Node* node);
+};
+
+// Constructor to initialize the BK Tree
```

Comment:
This comment is for the constructor. Please remove the new line between this comment and the constructor.